### PR TITLE
feat(Pipelines): validate parameter types upon PipelineVersion run

### PR DESCRIPTION
**Pull request by nazarfil**
## Changes

Enforces pipeline type arguments to contain only valid parameter types

## How/what to test

Try submitting pipeline version with invalid types

### CODE REVIEW

File: hexa/pipelines/graphql/schema.graphql [hunk 1]
```diff
@@ -36,10 +36,24 @@ enum PipelineRunStatus {
   terminating
 }
 
+enum ParameterTypes {
```
Comment (qgerome) [inline hexa/pipelines/graphql/schema.graphql:39]: The other enums are singular, I think it's better to keep the singular to be consistant.

File: hexa/pipelines/graphql/schema.graphql [hunk 2]
```diff
@@ -36,10 +36,24 @@ enum PipelineRunStatus {
...
+  s3
+  gcs
+  custom
+}
+
 type PipelineParameter {
   code: String!
   name: String!
-  type: String!
+  type: ParameterTypes
```
Comment (qgerome) [inline hexa/pipelines/graphql/schema.graphql:56]: `type` is required

**Review by qgerome — CHANGES_REQUESTED** [review]
It would be nice to have tests with incorrect types.

**Comment by qgerome** [comment]
Don't forget to edit the PR description to reflect your changes

**Comment by nazarfil** [comment]
Valid points I have updated the pr

**Review by qgerome — APPROVED** [review]

**Pull request merged**